src/librustc_mir/interpret/operand.rs

```diff
@@ -688,9 +688,6 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M>
         rval: OpTy<'tcx, M::PointerTag>,
     ) -> EvalResult<'tcx, (u128, usize)> {
         trace!("read_discriminant_value {:#?}", rval.layout);
-        if rval.layout.abi.is_uninhabited() {
-            return err!(Unreachable);
```

Comment:
Since this is unreachable (pun intended), can you change this into an assert?

Reply:
I am not sure if it is unreachable. But I am sure that the behavior we want is to treat such enums like all the others.